### PR TITLE
Support JSON-RPC

### PR DESCRIPTION
**Pull request by javiermanzo**

### CODE REVIEW

File: Example/HarborExample/ViewController.swift [hunk 1]
```diff
@@ -36,19 +39,35 @@ final class ViewController: UIViewController {
...
+//            case .success(let result):
+//                textAlert = result.quote
+//            case .cancelled:
+//                textAlert = "Request Cancelled"
+//            case .error:
+//                textAlert = "Request Error"
+//            }
+//            
+//            self.presentAlert(title: "Request Info", message: textAlert)
+//        }
```
Comment (OmarJalil) [inline Example/HarborExample/ViewController.swift:56]: I think the showcase should have both examples showcased

File: README.md [hunk 2]
```diff
@@ -208,60 +215,78 @@ let task = Task {
...
 
-To configure SSL Pinning, include the `sslPinningSHA256` property when initializing `HConfig` and configure Harbor with it:
+#### Set JSON RPC Version
+Use this method to set the JSON RPC version:
 
 ```swift
-let sslPinningSHA256 = "yourSHA256CertificateHash"
-let config = HConfig(sslPinningSHA256: sslPinningSHA256)
-Harbor.configure(config)
+HarborJRPC.setJRPCVersion("2.0")
```
Comment (OmarJalil) [inline README.md:257]:
```suggestion
// It uses 2.0 as default 
HarborJRPC.setJRPCVersion("2.0")
```

File: Sources/HarborJRPC/HJRPCRequestProtocol.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
+//
+//  File.swift
```
Comment (OmarJalil) [inline Sources/HarborJRPC/HJRPCRequestProtocol.swift:2]:
```suggestion
//  HJRPCRequestProtocol.swift
```

File: Sources/Harbor/HConfig.swift [hunk 4]
```diff
@@ -7,19 +7,9 @@
...
-        self.authProvider = authProvider
-        self.defaultHeaderParameters = defaultHeaderParameters
-        self.mTLS = mTLS
-        self.sslPinningSHA256 = sslPinningSHA256
-    }
+class HConfig {
+    var authProvider: HAuthProviderProtocol?
+    var defaultHeaderParameters: [String: String]?
+    var mTLS: HmTLS?
+    var sslPinningSHA256: String?
```
Comment (OmarJalil) [inline Sources/Harbor/HConfig.swift:14]: Why did you change it to a class?

Comment (javiermanzo) [reply Sources/Harbor/HConfig.swift]: I was trying another thing, changed again

File: Sources/HarborJRPC/HJSONRPCResponse.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    let id: String?
+    let result: Model?
+    let error: HJSONRPCError?
+}
+
+public struct HJSONRPCError: Codable, Sendable {
+    let code: Int
+    let message: String
+    let data: String?
+}
```
[the remark this answers is not in the document]
Comment (javiermanzo) [reply Sources/HarborJRPC/HJSONRPCResponse.swift]: Done

**Review by OmarJalil — APPROVED** [review]
LGTM <3
